### ISSUE TITLE
Panic on input that is not valid UTF8

### ISSUE THREAD

**Issue by dtolnay** [serde-rs/json#99]
This should be an error not a panic.

```rust
fn main() {
    let data = &[b'"', b'\xed', b'\xa0', b'\x80', b'"'];
    serde_json::from_slice::<String>(data).ok();
}
```

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Utf8Error { valid_up_to: 0 }'
```

Caused by [this unwrap](https://github.com/serde-rs/json/blob/e16bdbb28fbdcb85b0ee70f5ac51c210fe659de4/json/src/de.rs#L199).

**Comment by oli-obk** [comment]
we could generally enable the clippy lint for denying unwrap

**Comment by dtolnay** [comment]
I fixed the panic in #100 and I filed #101 to consider enabling the Clippy lint.
